typescript/sdk/src/deploy/core/HyperlaneCoreDeployer.ts

```diff
@@ -40,29 +44,104 @@ export class HyperlaneCoreDeployer<
     super(multiProvider, configMap, factoriesOverride, {
       logger: debug('hyperlane:CoreDeployer'),
     });
+    this.gasOverhead = objMap(configMap, (chain, config) => {
+      const { validators, threshold } = config.multisigIsm;
+      const verifyCost =
+        // @ts-ignore
+        multisigIsmVerifyCosts[`${validators.length}`][`${threshold}`];
+      if (!verifyCost)
+        throw new Error(
+          `Unknown verification cost for ${threshold} of ${validators.length}`,
+        );
+      return {
+        domain: ChainNameToDomainId[chain],
+        gasOverhead: verifyCost,
+      };
+    });
     this.startingBlockNumbers = objMap(configMap, () => undefined);
   }
 
-  deployInterchainGasPaymaster<LocalChain extends Chain>(
+  async deployInterchainGasPaymaster<LocalChain extends Chain>(
     chain: LocalChain,
+    deployerOwnedProxyAdmin: ProxyAdmin,
     proxyAdmin: ProxyAdmin,
     deployOpts?: DeployOptions,
   ): Promise<
     ProxiedContract<InterchainGasPaymaster, TransparentProxyAddresses>
   > {
-    return this.deployProxiedContract(
+    const interchainGasPaymaster = await this.deployProxiedContract(
       chain,
       'interchainGasPaymaster',
       [],
+      deployerOwnedProxyAdmin,
       proxyAdmin,
       [],
       deployOpts,
     );
+    return interchainGasPaymaster;
+  }
+
+  async deployDefaultIsmInterchainGasPaymaster<LocalChain extends Chain>(
+    chain: LocalChain,
+    interchainGasPaymasterAddress: types.Address,
+    deployOpts?: DeployOptions,
+  ): Promise<OverheadIgp> {
+    const chainSigner = this.multiProvider.getChainSigner(chain);
+    const deployer = await chainSigner.getAddress();
+    // Transfer ownership to the deployer so the destination gas overheads can be set
+    const initCalldata = Ownable__factory.createInterface().encodeFunctionData(
+      'transferOwnership',
+      [deployer],
+    );
+    const defaultIsmInterchainGasPaymaster = await this.deployContract(
+      chain,
+      'defaultIsmInterchainGasPaymaster',
+      [interchainGasPaymasterAddress],
+      {
+        ...deployOpts,
+        initCalldata,
+      },
+    );
```

Comment:
looking to make all of these deployments more atomic in future
eg config and transfer ownership in constructor so we don't have this awkwardness